ImageAnalysis/image_analysis/analyzers/UC_GenericLightSpecCam.py

```diff
@@ -130,6 +141,54 @@ def analyze_image(self, input_image: Array2D, auxiliary_data: Optional[dict] = N
                                                          input_parameters=self.build_input_parameter_dictionary())
         return return_dictionary
 
+    def perform_spectral_correction(self, wavelength, spectrum, image,
+                                    eff_low_bound=0.025):
+
+        # load detection efficiency
+        spectral_correction = self.construct_spectral_correction(wavelength)
+
+        # if any efficiency too low, don't correct
+        spectral_correction['efficiency'][spectral_correction['efficiency'] < eff_low_bound] = 1.0
+
+        # correct spectrum
+        cor_spectrum = spectrum / spectral_correction['efficiency']
+
+        # correct image
+        cor_image = image / spectral_correction['efficiency']
+
+        return cor_spectrum, cor_image
+
+    def construct_spectral_correction(self, input_wavelength):
+
+        # define relative path
+        rel_path = os.path.join('..', 'data')
```

Comment:
I'm not sure this works correctly - the path will likely be relative to the _current working directory_ rather than this python file. 
To make it explicit, use the special variable `__file__`: 
`rel_path = os.path.join(os.path.dirname(__file__), '..', 'data')`

or even better:
`spectral_correction_files_folder: Path = Path(__file__).parents[1] / 'data'`
using `from pathlib import Path`.